presto-jdbc/src/main/java/com/facebook/presto/jdbc/ConnectionParameters.java

```diff
@@ -0,0 +1,166 @@
+package com.facebook.presto.jdbc;
+
+import com.google.common.base.Splitter;
+import com.google.common.net.HostAndPort;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static io.airlift.http.client.HttpUriBuilder.uriBuilder;
+
+/**
+ * Container for parameters of a JDBC connection. The class is also responsible for parsing Presto JDBC URL.
+ */
+final class ConnectionParameters
```

Comment:
The class package-private controls access to the class (no one outside the package can use it). Everything that is part of the "public" API (for other users in the package) should be marked `public`, so it is clear what things are part of the public API.